### ISSUE TITLE
Sky TV Box

### ISSUE THREAD

**Issue by frankleggett** [bitfocus/companion-module-requests#1585]
I have asked before so I apologise for the repeat request. Is it possible for someone write a companion module for the Sky TV Box. A module that would change channel, Use the navigation arrows and possible show the TV Guide and exit.

There is already a Java and C# program that does this but it would be great to roll this up into a companion module then have control with a streamdeck.

SkyRemote programs on Github
https://github.com/seawright/SkyRemote

https://github.com/dalhundal/sky-remote-web

**Comment by peternewman** [comment]
Duplicate of #1393

As I mentioned in there, there's already an npm package available from dalhundal at https://github.com/dalhundal/sky-remote

**Comment by leonreucher** [comment]
[pkg.tgz](https://github.com/user-attachments/files/16715197/pkg.tgz)
Please give this a try and report if its working - you can also reach me on the Bitfocus Slack.

**Comment by thedist** [comment]
If you want to distribute a Companion module for testing, link the source repo, not packaged modules.

**Comment by leonreucher** [comment]
Yeah sorry for that, not the best idea to distribute unknown archives on here. I'll shift the communication with @frankleggett  to slack.

**Comment by thedist** [comment]
Slack or Github, either way I'd recommend they not touch a sketchy packaged module with no link to the source repo.

**Comment by leonreucher** [comment]
I agree on that - but tbh at the moment the whole GitHub and module publish flow makes it pretty hard for developers to quickly release new stuff. Many dev's recommended years ago, that the module base should be independent from the companion core and adding new repos should be a less manual process...

**Comment by thedist** [comment]
Takes 5 minutes to make your own repo, push your code to it, and then let users decide if they want to package from your source themselves, or use your package, or wait for it to be bundled with Companion.

I saw your package containing OpenTelemetry modules, and accessing environmental variables for databases and things like that, with a size greater than what would be needed for a Sky TV remote module, so that raises concerns. If everything is all above board then the source will show that, but it's difficult for users to see if it's legitimate when it's uglified and packaged.

**Comment by willosof** [comment]
Hmmm..

> at the moment the whole GitHub and module publish flow makes it pretty hard for developers to quickly release new stuff

Could you clarify what you mean by that, and making it an automated process? In what professional setting are you using Companion that makes you advocate for a direction where anyone can push anything with less oversight and control?

Even if many developers are advocating for something, it doesn’t necessarily mean it makes sense in the bigger picture. Feelings and what makes sense are often two very different things.

**Comment by leonreucher** [comment]
Duplicate of #1393
